### ISSUE TITLE
Tutorial update?

### ISSUE THREAD

**Issue by avaxman** [libigl/libigl#951]
Great work on 2.0!

However, the current tutorial compilation instructions would only compile 101. One needs to put in (per the warning given):

cmake -DCMAKE_BUILD_TYPE=Release ../ -DLIBIGL_WITH_OPENGL=ON -DLIBIGL_WITH_OPENGL_GLFW=ON

Perhaps these can be default?

#### Check all that apply (change to `[x]`)
- [ ] Windows
- [X] Mac OS X
- [ ] Linux

See <https://libigl.github.io/CONTRIBUTING/#bugreport> for more tips.

**Comment by jdumas** [comment]
If you compile the CMakelist at the root, it should build everything (tutorials included). We're trying to limit the amount of duplicated options, so the default behavior is OFF for everything in `libigl.cmake`, and `ON` for everything (except CGAL) in the root `CMakeLists.txt`. But I guess we probably need to update instructions on the website for that?

**Comment by avaxman** [comment]
Yes, please.

**Comment by avaxman** [comment]
Using "cmake -DCMAKE_BUILD_TYPE=Release ../../" will work (within "tutorial/build")

**Comment by alecjacobson** [comment]
While this will work, the new and recommend way to build everything is at the root directory ([soon to be added to docs](901f6e631fc12f212e1079d9fef414990194771c)).

```
cd libigl
mkdir build
cd build
cmake ../
```

**Comment by jdumas** [comment]
The only change we did as part of libigl 2.0 was use the official release tag for embree 2.17.4. Compared to commit cb61322, there is only ~10 days differences, but I guess they may have introduced some support for AVX512 that your machine doesn't support? (It did build on Travis and on our machines though). Can you try to update the [libigl cmake](https://github.com/libigl/libigl/blob/ed363da1f624e7a2f802f6b04973be8ea63f0511/cmake/LibiglDownloadExternal.cmake#L60) to use the same version of embree as before and see if it helps?

**Comment by avaxman** [comment]
Not sure exactly what to do...

**Comment by avaxman** [comment]
No, same error... (trying from a totally clean installation again). BTW, It's an iMac Late 2015 with High Sierra.

**Comment by avaxman** [comment]
Just to be clear, I changed the relevant lines into:
"
## Embree
function(igl_download_embree)
	igl_download_project(embree
		GIT_REPOSITORY https://github.com/embree/embree.git
		GIT_TAG        cb61322db3bb7082caed21913ad14869b436fe78
	)
endfunction()
"

**Comment by alecjacobson** [comment]
(btw, @avaxman if you put ``` before and after your copy-pastes then they won't get formatted weirdly.)

**Comment by avaxman** [comment]
It gives another error if you keep just compiling on:
```
/Users/amirvaxman/libigl/external/embree/kernels/common/scene.cpp:626:5: error: no member named 'createTriangleMesh' in namespace
      'embree::sse41'; did you mean simply 'createTriangleMesh'?
    SELECT_SYMBOL_DEFAULT_AVX(device->enabled_cpu_features,createTriangleMesh);
```
It looks like there is some namespace problem with both, these functions seem to exist.

**Comment by avaxman** [comment]
BTW (sorry for being annoying..), but the cmake has this:

```
# (by default, static build is off since libigl is a header-only library)
option(LIBIGL_USE_STATIC_LIBRARY "Use LibIGL as static library" ON)
```
Shouldn't it then be OFF?

**Comment by alecjacobson** [comment]
haha, oops. @danielepanozzo I think we converged on `ON` but didn't fix the comment. Is that correct?

**Comment by jdumas** [comment]
It should be `ON` in the root CMakeLists.txt (to catch missing template instantiations on Travis), and `OFF` in the `libigl.cmake` (for user to be able to use libigl however they want). When the new python bindings come will be a different story, since they need libigl compiled header-only (because of too many template instantiations).

@avaxman what happens if you compile the tutorials and embree in the libigl-legacy repository? Do you still get the same issue? Otherwise, can you try to pass `-DEMBREE_ISA_AVX512SKX=OFF` to the CMake (see discussion in #947), that maybe we could set to `OFF` by default.

**Comment by avaxman** [comment]
hmm, actually legacy gives the same error (which makes sense providing I reverted the embree version, and that is what i throwing the error). I might have been using much older version in my current works.

Maybe this will shed some light: I noticed the cmake outputs:

```
-- Detecting default ISA:
-- Detecting default ISA: SSE2
```

**Comment by jdumas** [comment]
Ok. There are actually a couple of [CMake options](https://github.com/embree/embree/blob/09a6013b79d0a670ca630d121b86265f9b9fab99/CMakeLists.txt#L239) you can play with to enable/disable SSE or AVX. You can also try the PR #947 that updates embree to version 3.x, see if you find a way to compile on your machine. Maybe also try to compile embree standalone (as the error may appear only when embree is included as a subdirectory). Would be nice if we could keep SSE/AVX `ON` by default (or at least correctly detected by CMake). Please let us know if you find a combination that works on your machine! (What hardware do you have?)

**Comment by avaxman** [comment]
So far nothing really helps. my system is as follows
<img width="529" alt="screen shot 2018-10-17 at 18 28 50" src="https://user-images.githubusercontent.com/8985720/47101503-89678880-d23a-11e8-8df0-4ad44da839f6.png">

**Comment by jdumas** [comment]
It's a stupid remark, but when you disable all the options, you clear your `CMakeCache.txt` right? (Or alternatively sets the option using `ccmake ..`). Without the same hardware, it's gonna be difficult for me to reproduce unfortunately =/

**Comment by avaxman** [comment]
I am getting the same errors out of the box with another OsX, this time this:
<img width="559" alt="screen shot 2018-10-17 at 20 22 02" src="https://user-images.githubusercontent.com/8985720/47107756-6cd34c80-d24a-11e8-8f44-a645480297da.png">
Maybe it's an old Mac thing?
However, perhaps something helpful: if I cmake it for XCode, it seems to be working. However, another problem arises:
```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip -x /Users/amirvaxman/libigl/python/Debug/pyigl.so
strip -u -r /Users/amirvaxman/libigl/python/pyigl.so
error: strip: can't open file: /Users/amirvaxman/libigl/python/pyigl.so (No such file or directory)
make: *** [pyigl_buildpart_1] Error 1
```
But the individual tutorial projects are working!

**Comment by jdumas** [comment]
We're working on new python bindings right so I wouldn't spend too much time fixing issues with the current one =)

**Comment by tbarton16** [comment]
@avaxman Can you post/link instructions for making using Xcode? I am having the same error
